diff --git a/dist/shared/better-auth.C3-_8m-g.mjs b/dist/shared/better-auth.C3-_8m-g.mjs
index 838c127ad348b4673e1744a47d43d03bcf6d8172..857a22bd438e79d3c0392a966f136981a8895c26 100644
--- a/dist/shared/better-auth.C3-_8m-g.mjs
+++ b/dist/shared/better-auth.C3-_8m-g.mjs
@@ -6407,9 +6407,8 @@ const getAccessToken = createAuthEndpoint(
       let newTokens = null;
       const accessTokenExpired = account.accessTokenExpiresAt && new Date(account.accessTokenExpiresAt).getTime() - Date.now() < 5e3;
       if (account.refreshToken && accessTokenExpired && provider.refreshAccessToken) {
-        newTokens = await provider.refreshAccessToken(
-          account.refreshToken
-        );
+        const refreshToken = await decryptOAuthToken(account.refreshToken, ctx.context);
+        newTokens = await provider.refreshAccessToken(refreshToken);
         await ctx.context.internalAdapter.updateAccount(account.id, {
           accessToken: await setTokenUtil(newTokens.accessToken, ctx.context),
           accessTokenExpiresAt: newTokens.accessTokenExpiresAt,
@@ -6418,10 +6417,8 @@ const getAccessToken = createAuthEndpoint(
         });
       }
       const tokens = {
-        accessToken: await decryptOAuthToken(
-          newTokens?.accessToken ?? account.accessToken ?? "",
-          ctx.context
-        ),
+        accessToken: newTokens?.accessToken ??
+          (await decryptOAuthToken(account.accessToken ?? "", ctx.context)),
         accessTokenExpiresAt: newTokens?.accessTokenExpiresAt ?? account.accessTokenExpiresAt ?? void 0,
         scopes: account.scope?.split(",") ?? [],
         idToken: newTokens?.idToken ?? account.idToken ?? void 0
